app/src/main/java/com/pb/nex26dv/shippingapp/WebService/FileDownloader.java

```diff
@@ -0,0 +1,46 @@
+package com.pb.nex26dv.shippingapp.WebService;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * Created by NEX9ZKA on 12/15/2016.
+ */
+
+public class FileDownloader {
+    private static final int MEGABYTE = 1024 * 1024;
+
+    public static void downloadFile(String fileUrl, File directory) {
+        try {
+
+            URL url = new URL(fileUrl);
+            HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+            //urlConnection.setRequestMethod("GET");
+            //urlConnection.setDoOutput(true);
+            urlConnection.connect();
+
+            InputStream inputStream = urlConnection.getInputStream();
+            FileOutputStream fileOutputStream = new FileOutputStream(directory);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fileOutputStream*. The resource is referenced in statements at the following line: 35. The resource closure statement is at line: 37. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by InputStream.read. Either a) close *fileOutputStream* in a try-finally block or b) close the resource by declaring *fileOutputStream* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).